### DISCUSSION TITLE
Limit of IFS Browser: how many files can it display?

### DISCUSSION THREAD

**Discussion by GagomeSmirnja** [codefori/vscode-ibmi#2246] [General]
I am using Code for i v2.13.
And I have noticed that when I try to go to a directory which has many files it doesn't work. I see the animation that it is getting data from the server and after some tiime (about 20 seconds) the animation stops and nothing is happening. I would expect that the directory is opened and I see the content. 
Now when I go to a shell and do a simple ls -1 | wc -l in this directory I see there are more than 100000 files. 
Is there a limit build in in the IFS Browser? How many files can it handle?

**Comment by chrjorgensen** [comment]
@GagomeSmirnja With over 100,000 files in one directory I don't think you would like to have to scroll through all those files... Not the right tool for the job...

I would recommend install [Midnight Commander](https://en.wikipedia.org/wiki/Midnight_Commander) on your system - a Norton Commander-like file manager. It has some filtering capabilities and is running directly on the server, not as a client on another computer.

```bash
yum install mc
```

**Reply by GagomeSmirnja** [reply]
Thank you.
Yes, I know with more than 100000 files it is difficult to use the browser. All I wanted to enter a subdirectory below and not scroll through all the files. But the way the IFS Browser reacted is not OK I think. It should have sent some kind of error message "to many files" so that I know what is happening. But instead it did nothing and I didn't know what was wrong.
I added a shortcut to the subdirectory to which I wanted to go and now it works for me and I can use the IFS Browser to edit the file I wanted.
But maybe some kind of better error handling would be nice to have.
Thanks.

**Reply by SJLennon** [reply]
Just wool gathering...  For these "unusual" situations, perhaps what is needed it the ability to create a shortcut that shows only subdirectories. I suppose it would then need a right click to show the contents of the selected subdirectory.

**Reply by chrjorgensen** [reply]
The tooltip for the directory uses 'ls | wc -l' and may succeed in showing the number of files when hovering over the directory, before opening the directory...?

**Reply by GagomeSmirnja** [reply]
When I hover with the mouse over a directory all I get is the full path name in the tooltip. There is no information about files. 
I see this with every directory.
Do I have to setup something to get "ls | wc -l"?

**Reply by chrjorgensen** [reply]
@GagomeSmirnja Sorry, my mistake - the `ls | wc -l` is used for source files in the object browser, not in the IFS browser... 😞
But maybe an idea to have it in the IFS browser as well?
